x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/AbstractGeoHexGridTiler.java

```diff
@@ -48,10 +57,9 @@ public int setValues(GeoShapeCellValues values, GeoShapeValues.GeoShapeValue geo
         assert bounds.minX() <= bounds.maxX();
         // first check if we are touching just fetch cells
         if (bounds.maxX() - bounds.minX() < 180d) {
-            final long minH3 = H3.geoToH3(bounds.minY(), bounds.minX(), precision);
-            final long maxH3 = H3.geoToH3(bounds.maxY(), bounds.maxX(), precision);
-            if (minH3 == maxH3) {
-                return setValuesFromPointResolution(minH3, values, geoValue);
+            final long singleCell = boundsInSameCell(bounds, precision);
```

Comment:
Can we add here the constraint that precision > 0. We are doing that check later on.